### PR TITLE
Update zboss libraries to v3.5.2.0

### PR DESCRIPTION
**Pull request by tomchy**
* Providing the two sets of libraries allowed us to add experimental features with the production version intact.
After the NCS v1.5.1 release, there will be only a single library to reduce the complexity of the Zigbee stack configuration.
* Update ZBOSS libraries to v3.5.2.0

Signed-off-by: Tomasz Chyrowicz <tomasz.chyrowicz@nordicsemi.no>

### CODE REVIEW

**Review by maciekfabia — APPROVED** [review]

**Review by rlubos — COMMENTED** [review]
Just some nits

File: zboss/CMakeLists.txt [hunk 1]
```diff
@@ -1,8 +1,103 @@
 #
-# Copyright (c) 2021 Nordic Semiconductor
+# Copyright (c) 2020 Nordic Semiconductor ASA
```
[the remark this answers is not in the document]
Comment (tomchy) [reply zboss/CMakeLists.txt:2]: We've dropped the variant, in which both creation and modification date is present. That's why only the 2020 is here.

**Review by rlubos — APPROVED** [review]

File: zboss/include/osif/zb_macll.h [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021 Nordic Semiconductor ASA
+ * Copyright (c) 2020 Nordic Semiconductor ASA
```
[the remark this answers is not in the document]
Comment (tomchy) [reply zboss/include/osif/zb_macll.h:2]: That's a bit strange - probably a mistake in the previous library update. This file was present in 2020.

File: zboss/doc/zboss_configuration.rst [hunk 3]
```diff
@@ -8,67 +8,21 @@ ZBOSS library configuration
    :depth: 2
 
 The ZBOSS stack is distributed as a set of precompiled linkable libraries.
-These libraries are available in the following versions:
 
-* `ZBOSS production libraries`_
-* `ZBOSS development libraries`_
-
-Each version contains variants of the libraries that support different Zigbee device roles, each variant having its own configuration.
+The library support different Zigbee device roles, each variant having its own configuration.
```
Comment (greg-fer) [inline zboss/doc/zboss_configuration.rst:12]:
```suggestion
The libraries support different Zigbee device roles, with each variant having its own configuration.
```

File: zboss/doc/zboss_configuration.rst [hunk 4]
```diff
@@ -8,67 +8,21 @@ ZBOSS library configuration
...
-* :file:`zboss/production/include/osif/libzboss_config.h` - library for Coordinators and Routers
-* :file:`zboss/production/include/osif/libzboss_config.ed.h` - library for End Devices
-
-|list_outro|
-
-The ZBOSS production library version is enabled by default with the :option:`CONFIG_ZIGBEE_LIBRARY_PRODUCTION` Kconfig option.
-The production libraries support only the nRF52840 and nRF52833.
-
-ZBOSS development libraries
+ZBOSS libraries
```
Comment (greg-fer) [inline zboss/doc/zboss_configuration.rst:15]: ZBOSS device role libraries? Or maybe we don't need this section header at all?

File: zboss/doc/zboss_configuration.rst [hunk 5]
```diff
@@ -8,67 +8,21 @@ ZBOSS library configuration
...
+* :file:`zboss/include/osif/libzboss_config.h` - library for Coordinators and Routers
+* :file:`zboss/include/osif/libzboss_config.ed.h` - library for End Devices
 
 |list_outro|
 
-You can select the ZBOSS development library version with the :option:`CONFIG_ZIGBEE_LIBRARY_DEVELOPMENT` Kconfig option.
-You can enable additional features in the development version of ZBOSS libraries using the following Kconfig options:
+You can enable additional features in the ZBOSS libraries using the following Kconfig options:
 
 * :option:`CONFIG_ZIGBEE_LIBRARY_NCP_DEV` - With this option enabled, the application may link NCP commands implementation.
```
Comment (greg-fer) [inline zboss/doc/zboss_configuration.rst:27]:
```suggestion
* :option:`CONFIG_ZIGBEE_LIBRARY_NCP_DEV` - With this option enabled, the application includes the implementation of NCP commands.
```

File: zboss/doc/zboss_configuration.rst [hunk 6]
```diff
@@ -80,7 +34,7 @@ Both of these features are experimental.
 ZBOSS library certification
 ***************************
 
-Zigbee ZBOSS libraries in |NCS| in the production state are always tested in-house by Nordic Semiconductor to conform to Zigbee Pro R22 test specification.
+Zigbee ZBOSS libraries in |NCS| are always tested in-house by Nordic Semiconductor to conform to Zigbee Pro R22 test specification.
```
Comment (greg-fer) [inline zboss/doc/zboss_configuration.rst:37]:
```suggestion
Zigbee ZBOSS libraries in the |NCS| are always tested in-house by Nordic Semiconductor to conform to Zigbee Pro R22 test specification.
```

File: zboss/doc/zboss_configuration.rst [hunk 7]
```diff
@@ -80,7 +34,7 @@ Both of these features are experimental.
...
 However, these libraries are only periodically verified by external test service provider and granted Zigbee Compliant Platform certification by Zigbee Alliance.
 
 .. |list_intro| replace:: For a complete list of the ZBOSS configuration options, see the following files:
```
Comment (greg-fer) [inline zboss/doc/zboss_configuration.rst:40]: Please change |list_outro| to the following:
``These libraries are used in the Zigbee protocol configuration in the |NCS| when defining the Zigbee device role, as described in :ref:`nrf:zigbee_ug_configuration` in the |NCS| documentation.``

**Review by greg-fer — CHANGES_REQUESTED** [review]
ZBOSS API docs need to be updated when this PR is merged. Let's discuss f2f.

File: zboss/doc/zboss_configuration.rst (hunk 4 above)
Comment (tomchy) [reply zboss/doc/zboss_configuration.rst:15]: removed

File: zboss/doc/zboss_configuration.rst (hunk 3 above)
Comment (tomchy) [reply zboss/doc/zboss_configuration.rst]: Proposed change applied.

File: zboss/doc/zboss_configuration.rst (hunk 5 above)
Comment (tomchy) [reply zboss/doc/zboss_configuration.rst]: Proposed change applied.

File: zboss/doc/zboss_configuration.rst (hunk 7 above)
Comment (tomchy) [reply zboss/doc/zboss_configuration.rst:37]: Proposed change applied.

**Review by greg-fer — APPROVED** [review]
Bring it on.

**Pull request merged**